### PR TITLE
 [DNNL] Add bfloat16 type support for dnnl conv2d kernel

### PR DESCRIPTION
**Pull request by Qianshui-Jiang**
Added bfloat16 for dnnl conv2d kernel in order to coordinate with bf16/fp32 mixed precision mode, 
which implemented ny changing the type of dnnl primitive descriptor & memory descripor accoring to src/dst data format during the declaration.

Test case also modified for both bf16 & fp32.

### CODE REVIEW

File: python/tvm/contrib/dnnl.py [hunk 1]
```diff
@@ -113,6 +113,17 @@ def dnnl_conv2d(
...
+    pre_cast = False
+    post_cast = False
+    if src.dtype == "float32":
+        pre_cast = True
+    elif src.dtype == "bfloat16":
+        pre_cast = False
+    if out_dtype == "float32":
+        post_cast = True
+    elif out_dtype == "bfloat16":
+        post_cast = False
```
Comment (comaniac) [inline python/tvm/contrib/dnnl.py:125]:
```suggestion
    pre_cast = src.dtype == "float32"
    post_cast = out_dtype == "float32"
```

**Review by comaniac — APPROVED** [review]
LGTM. Just a nit.

**Comment by Qianshui-Jiang** [comment]
> LGTM. Just a nit.

thx a lot !

**Comment by comaniac** [comment]
Thanks @Qianshui-Jiang

**Pull request merged**